client/client.go

```diff
@@ -488,6 +490,73 @@ func (c *client) GetRegion(ctx context.Context, key []byte) (*Region, error) {
 	return c.parseRegionResponse(resp), nil
 }
 
+func (c *client) waitForRegion(ctx context.Context, key []byte, memberURLs []string) (*pdpb.GetRegionResponse, error) {
+	childContext, cancel := context.WithCancel(ctx)
+
+	ch := make(chan *pdpb.GetRegionResponse, len(memberURLs))
+
+	for _, memberURL := range memberURLs {
+		go c.getRegionResponse(childContext, memberURL, key, ch)
```

Comment:
As @nolouch said, you could try to get region from these members one by one and return the first valid response. Doing this concurrently will cause read amplification.